Cyb3rWard0g/HELK
# WDATP data ingestion

**What is this PR for?**
WDATP ingestion 
**What type of PR is it?**
[Bug Fix] or [Feature Request]
Freature request
**How should this be tested?**
Specific steps, configs and images if available

WDATP subciption is needed and specific steps has to be done prior to using this 

these are outlined in the links in the comment section of the logstash filter files
[2513-winevent-wdatp-conversion-filter.conf.txt](https://github.com/Cyb3rWard0g/HELK/files/4070293/2513-winevent-wdatp-conversion-filter.conf.txt)
[2514-winevent-wdatp-filter.conf.txt](https://github.com/Cyb3rWard0g/HELK/files/4070294/2514-winevent-wdatp-filter.conf.txt)

im not much of a github user so, if there is a more githubby way of submitting these files to you i aplogise ahead of time.

one caveats i have is that currently i did not include the WDATP specific dashboards, for kibana. just the filter files.

during my use of this i have observed some chunks/ stopping of data flow from azure to the logstash plugin. im trying to figure out the rootcause of this but have yet to find a final solution. i may be related to MS patching or what ever they do to the AZure infrastructure. 

however once the data stream resumes, the data does catch op - alt least there is a huge peak in data being ingested. It may be an idea to index based on the event recorded - data filed instead of the standard elastisc timestamp. i currently se about 5-7 min delay in data being indexed. Its not at fast as a onprem helk, but the fact that i do not have to hassle with tuning sysmon on our clients to ensure acceptable performance impact i still see value in this.

i have also added a local ip traffic topology filter, that i use to easy detect ex client to client traffic. details of usage is in the comments within the file.

I 
[8200-local-ip-filter.conf.txt](https://github.com/Cyb3rWard0g/HELK/files/4070355/8200-local-ip-filter.conf.txt)
 



**Questions:**
* Do the licenses files need update? No
* Are there breaking changes for older versions? No
* Does this needs documentation? No
